### PR TITLE
Added revised Case Finding report and Treatment Outcome report

### PR DESCRIPTION
**Pull request by kkrampa**
@czue 

treatment outcome

case finding

### CODE REVIEW

File: custom/enikshay/reports/case_finding.py
Comment (czue) [inline custom/enikshay/reports/case_finding.py]: nit: it might be nice if these columns were automatically generated from `AGE_RANGES`

File: custom/enikshay/__init__.py
Comment (czue) [inline custom/enikshay/__init__.py]: I think we are trying to move these into `reports.py` moving forwards to avoid too many imports in the `__init__` modules

File: custom/enikshay/reports/filters.py
Comment (czue) [inline custom/enikshay/reports/filters.py]: is there anything that makes this a custom filter as opposed to one that could be dropped into any report for locations?

File: custom/enikshay/reports/views.py
Comment (czue) [inline custom/enikshay/reports/views.py]: do we not have anything like this already? i thought we had some other location APIs we might be able to use / modify. cc @esoergel

File: custom/enikshay/templates/enikshay/partials/report_content.html
Comment (czue) [inline custom/enikshay/templates/enikshay/partials/report_content.html]: is there a reason why none of this could be pulled out/shared with [tabular.html](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/templates/reports/async/tabular.html#L73) ?

**Review by czue — COMMENTED** [review]
thanks for putting this together!

no blockers but a couple comments i'll let you respond on before merge

File: custom/enikshay/reports/views.py
Comment (kkrampa) [inline custom/enikshay/reports/views.py]: We have a similar view for UCR filters but it takes as argument ```report_id``` so I copied and modified it a little bit.

File: custom/enikshay/reports/filters.py
Comment (czue) [inline custom/enikshay/reports/filters.py]: sorry, probably an obvious question but why can't you use the UCR filter? do you mean the `LocationChoiceProvider`?

**Comment by czue** [comment]
@proteusvacuum can you do a pass on this? just wondering if there are any issues/conflicts with the location filtering stuff you're working on.

File: custom/enikshay/reports/filters.py
Comment (proteusvacuum) [inline custom/enikshay/reports/filters.py]: Yeah, would be good to make this inherit from `LocationChoiceProvider` so that it also gets the location restriction features I'm currently working on (which are required for these reports)

File: custom/enikshay/templates/enikshay/partials/report_content.html
Comment (kkrampa) [inline custom/enikshay/templates/enikshay/partials/report_content.html]: I changed it in 2acefe5 and a8be8b1

File: custom/enikshay/reports/views.py
Comment (proteusvacuum) [inline custom/enikshay/reports/views.py]: Can you inherit from the view that we are currently using? I'm currently making changes to it so that it works with location restrictions. This view will get blocked by the location middleware otherwise.

**Comment by proteusvacuum** [comment]
@kkrampa I'm currently working on restricting the locations in UCR reports so that people only see data based on their assigned locations. I should have something pushed later today and will ping you on it so you can see what I've done. Both the filter and the locations view that you've added here will need to be changed slightly (or inherit from the ones we currently use) so that this functionality will work.

File: custom/enikshay/reports/views.py
Comment (kkrampa) [inline custom/enikshay/reports/views.py]: In UCRs this view https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/views.py#L1220 is used and it's not Class Based View.

**Comment by kkrampa** [comment]
@proteusvacuum So then, I will wait for your changes.

**Comment by czue** [comment]
@kkrampa see https://github.com/dimagi/commcare-hq/pull/13476

**Comment by kkrampa** [comment]
@proteusvacuum @czue 
It seems I only need to pass user here https://github.com/dimagi/commcare-hq/pull/13446/files#diff-8502d8a84c3efa0a2c8abd6efe412327R21 to make this work with @proteusvacuum changes

**Comment by proteusvacuum** [comment]
@kkrampa, you'll also need to update https://github.com/dimagi/commcare-hq/pull/13446/files#diff-49ee5c39504653740388978724eec38dR20 to only include locations that are accessible to the user, I think. I think it might be better if you used the UCR `LocationChoiceProvider` filter in that case... I'm not sure I understand why it can't be used in this instance. 
You should also decorate all the new views you have with the `@location_safe` decorator.

File: custom/enikshay/reports/views.py
Comment (esoergel) [inline custom/enikshay/reports/views.py]: @czue the closest thing I can think of is the `EmwfOptionsView`, but that also handles users and groups, so it probably doesn't make sense.  There's the tastypie location resource, but that doesn't currently accept a query parameter.

**Comment by kkrampa** [comment]
@proteusvacuum I will do it.
What's the status of #13476? Are you going to merge this soon?

**Comment by proteusvacuum** [comment]
@kkrampa thanks. That PR should be merged sometime early next week once @esoergel signs off on it.

File: custom/enikshay/reports/sqldata/case_finding_sql_data.py
Comment (proteusvacuum) [inline custom/enikshay/reports/sqldata/case_finding_sql_data.py]: should this be `suspect`?

File: settings.py
Comment (proteusvacuum) [inline settings.py]: does this mean we can remove the `custom.enikshay.integrations.ninetyninedots` line above?

File: custom/enikshay/reports/views.py
Comment (proteusvacuum) [inline custom/enikshay/reports/views.py]: @esoergel do you know if this will work with your new method decorator? Or should we wait for your PR to be merged?

**Review by proteusvacuum — CHANGES_REQUESTED** [review]

**Comment by proteusvacuum** [comment]
Thanks for the updates @kkrampa. I had a few more minor comments, but the crux of this looks great.

File: custom/enikshay/reports/views.py
Comment (esoergel) [inline custom/enikshay/reports/views.py]: Looks like you just merged the other PR, so @kkrampa can pull in master and use that version.  I haven't tested this wrapped with `@method_decorator`, but it should work out of the box as a method decorator on a dispatch method or as a class decorator on a django view.

My preference would be:
```python
@location_safe
class LocationsView(View):
    ...
```
As is now done for the other class-based views.

File: custom/enikshay/reports/generic.py
Comment (kkrampa) [inline custom/enikshay/reports/generic.py]: It wasn't supposed to be a stub but nonetheless, I've changed a name of this variable.

**Review by proteusvacuum — APPROVED** [review]

**Comment by proteusvacuum** [comment]
:+1: on :white_check_mark:

**Comment by kkrampa** [comment]
@proteusvacuum :white_check_mark:

**Pull request merged**
